pkg/resourceinterpreter/defaultinterpreter/healthy.go

```diff
@@ -32,8 +32,16 @@ func interpretDeploymentHealth(object *unstructured.Unstructured) (bool, error)
 		return false, err
 	}
 
-	healthy := (deploy.Status.UpdatedReplicas == *deploy.Spec.Replicas) && (deploy.Generation == deploy.Status.ObservedGeneration)
-	return healthy, nil
+	if deploy.Generation != deploy.Status.ObservedGeneration {
+		return false, nil
+	}
+	if deploy.Status.UpdatedReplicas < *deploy.Spec.Replicas {
+		return false, nil
+	}
```

Comment:
```suggestion
	if deploy.Status.ObservedGeneration != deploy.Generation {
		return false, nil
	}
	if (deploy.Spec.Replicas != nil) && deploy.Status.UpdatedReplicas < *deploy.Spec.Replicas {
		return false, nil
	}
```
My suggestion:
1. put `deploy.Status` ahead of `deploy.Generation`
2. make sure `deploy.Spec.Replicas` not nil before dereference. (I know it might never happen, just to avoid false-positive warnings from the static checks.)

same as below